web/client/components/widgets/builder/wizard/map/enhancers/handleMapSelect.js

```diff
@@ -26,17 +26,24 @@ const handleMapSelect = compose(
                 return {
                     ...(mapState && mapState.map || {}),
                     ...(includeMapId ? {id: map.id} : {}),
+                    groups: mapState && [...mapState.groups] || [],
```

Comment:
Maybe this is more robust (in case groups === undefined), from some old maps.
```suggestion
                    groups: mapState && mapState.groups || [],
```